cocos/2d/CCNode.cpp

```diff
@@ -1714,7 +1725,17 @@ const Mat4& Node::getNodeToParentTransform() const
         //move to anchor point first, then rotate
         Mat4::createTranslation(x, y, z, &translation);
         
-        Mat4::createRotation(_rotationQuat, &_transform);
+        _transform = Mat4::IDENTITY;
+
+        if (_isRelativeToWorld && _parent)
+        {
+            _transform = _transform * _parent->getWorldToNodeTransform();
+        }
+
+        Mat4 rotation;
+        Mat4::createRotation(_rotationQuat, &rotation);
+
+        _transform = _transform * rotation;
```

Comment:
The function name is `getNodeToParentTransform `, so i think the logic is not good here.

Reply:
Ok. Other than this, do you think the feature could be acceptable into Node?

Reply:
I think the game logic should handle it manually, not add the feature in Node. Node is now very heavy, it is better no add new features there.